File: prototype/rails/src/app/domain/shop/status_label_test.rb
require "minitest/autorun"
require_relative "status_label"

class StatusLabelTest < Minitest::Test
  StatusLabel = Domain::Shop::StatusLabel

  def test_it_reads_a_stored_status_as_a_sentence
    assert_equal "Awaiting shipment", StatusLabel.humanize("awaiting_shipment")
  end

  def test_a_single_word_status_keeps_its_shape
    assert_equal "Paid", StatusLabel.humanize("paid")
  end
end

File: prototype/php/src/app/Domain/Shop/StatusLabelTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Shop;

use PHPUnit\Framework\TestCase;

final class StatusLabelTest extends TestCase
{
    public function test_it_reads_a_stored_status_as_a_sentence(): void
    {
        $this->assertSame('Pending verification', StatusLabel::humanize('pending_verification'));
        $this->assertSame('Paid', StatusLabel::humanize('paid'));
    }
}
